kafka-scheduler/src/main/java/com/mesosphere/dcos/kafka/plan/KafkaUpdatePhase.java

```diff
@@ -3,39 +3,39 @@
 import com.mesosphere.dcos.kafka.config.KafkaSchedulerConfiguration;
 import com.mesosphere.dcos.kafka.offer.KafkaOfferRequirementProvider;
 import com.mesosphere.dcos.kafka.state.FrameworkState;
-import org.apache.mesos.scheduler.ChainedObserver;
-import org.apache.mesos.scheduler.plan.Block;
-import org.apache.mesos.scheduler.plan.Phase;
+import org.apache.mesos.scheduler.plan.DefaultPhase;
+import org.apache.mesos.scheduler.plan.Step;
+import org.apache.mesos.scheduler.plan.strategy.SerialStrategy;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
 
-public class KafkaUpdatePhase extends ChainedObserver implements Phase {
-  private final List<Block> blocks;
-  private final String configName;
-  private final KafkaSchedulerConfiguration config;
-  private final UUID id;
+public class KafkaUpdatePhase extends DefaultPhase {
 
-  public KafkaUpdatePhase(
+  public static KafkaUpdatePhase create(
       String targetConfigName,
       KafkaSchedulerConfiguration targetConfig,
       FrameworkState frameworkState,
       KafkaOfferRequirementProvider offerReqProvider) {
-    this.configName = targetConfigName;
-    this.config = targetConfig;
-    this.blocks = createBlocks(configName, config.getServiceConfiguration().getCount(), frameworkState, offerReqProvider);
-    this.id = UUID.randomUUID();
+
+      return new KafkaUpdatePhase(targetConfigName,
+                createBlocks(targetConfigName,
+                        targetConfig.getServiceConfiguration().getCount(),
+                        frameworkState, offerReqProvider)
+      );
+  }
+  public KafkaUpdatePhase(String name,  List<Step> steps){
+          super(name, steps, new SerialStrategy<>(), new ArrayList<>());
   }
 
-  @Override
-  public List<Block> getBlocks() {
-    return blocks;
+  public List<Step> getBlocks() {
+    return getChildren();
   }
 
-  @Override
-  public Block getBlock(UUID id) {
-    for (Block block : getBlocks()) {
+
+  public Step getBlock(UUID id) {
```

Comment:
This is going to be really confusing down the road when we're calling things `block` even though they have type `Step`, could we change the name of the method and the variable names?